File: src/1431-Kids-With-the-Greatest-Number-of-Candies/kids_with_the_greatest_number_of_candies.py
from typing import List

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        ans: List[bool] = []
        h: int = max(candies)
        
        for i in range(len(candies)):
            if candies[i] + extraCandies >= h:
                ans.append(True)
            else:
                ans.append(False)

        return ans
    
if __name__ == "__main__":
    candies: List[int] = [2,3,5,1,3]
    extra_candies: int = 3

    solution = Solution()

    print(solution.kidsWithCandies(candies, extra_candies))

File: src/1431-Kids-With-the-Greatest-Number-of-Candies/kids_with_the_greatest_number_of_candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
  let ans = []
  let h = Math.max(...candies)

  for (let i = 0; i < candies.length; i++) {
    if (candies[i] + extraCandies >= h) {
      ans.push(true)
    }
    else {
      ans.push(false)
    }
  }

  return ans
};

let candies = [2,3,5,1,3]
let extra_candies = 3

console.log(kidsWithCandies(candies, extra_candies))